samples/nrf9160/gps/src/supl_support.c

```diff
@@ -0,0 +1,154 @@
+#include <zephyr.h>
+#include <supl_session.h>
+#include <stdio.h>
+#include <logging/log.h>
+
+#define SUPL_SERVER "supl.google.com"
+
+static int            supl_fd;
+
+int open_supl_socket(void)
+{
+	int err;
+	int proto;
+	u16_t port;
+	struct nrf_addrinfo *addr;
+	struct nrf_addrinfo *info;
+
+	proto = NRF_IPPROTO_TCP;
+	port = nrf_htons(7276);
+
+	struct nrf_addrinfo hints = {
+		.ai_family = NRF_AF_INET,
+		.ai_socktype = NRF_SOCK_STREAM,
+		.ai_protocol = proto,
+		/* Either a valid,
+		 * NULL-terminated access point name or NULL.
+		 */
+		.ai_canonname = NULL,
+	};
+
+	err = nrf_getaddrinfo(SUPL_SERVER, NULL, &hints, &info);
+	if (err) {
+		printk("Failed to resolve hostname %s on IPv4, errno: %d)\n",
+			log_strdup(SUPL_SERVER),
+			errno);
+		return -1;
+	}
+
+	supl_fd = nrf_socket(NRF_AF_INET, NRF_SOCK_STREAM, proto);
+	if (supl_fd < 0) {
+		printk("Failed to create socket, errno %d\n", errno);
+		goto cleanup;
+	}
+
+	struct nrf_timeval timeout = {
+		.tv_sec = 1,
+		.tv_usec = 0,
+	};
+
+	err = nrf_setsockopt(supl_fd,
+			     NRF_SOL_SOCKET,
+			     NRF_SO_RCVTIMEO,
+			     &timeout,
+			     sizeof(timeout));
+	if (err) {
+		printk("Failed to setup socket timeout, errno %d\n", errno);
+		return -1;
+	}
+
+	/* Not connected */
+	err = -1;
+
+	for (addr = info; addr != NULL; addr = addr->ai_next) {
+		struct nrf_sockaddr *const sa = addr->ai_addr;
+
+		switch (sa->sa_family) {
+		case NRF_AF_INET6:
+			((struct nrf_sockaddr_in6 *)sa)->sin6_port = port;
+			break;
+		case NRF_AF_INET:
+			((struct nrf_sockaddr_in *)sa)->sin_port = port;
+			char ip[255] = { 0 };
+
+			nrf_inet_ntop(NRF_AF_INET,
+				      (void *)&((struct nrf_sockaddr_in *)
+					      sa)->sin_addr,
+				      ip,
+				      255);
+			printf("ip %s (%x) port %d\n",
+				log_strdup(ip),
```

Comment:
You sholud not do this with `printk`

Reply:
ok

Reply:
Still not fixed, remove `log_strdup` here and grep for `printf` -> `printk`

Reply:
fixed now